DQM/Phase2OuterTracker/plugins/OuterTrackerMonitorTTCluster.cc

```diff
@@ -0,0 +1,389 @@
+  std::string HistoName;
+  
+  dqmStore_->setCurrentFolder(topFolderName_+"/Clusters/NClusters");
+  
+  // NClusters
+  edm::ParameterSet psTTCluster_Barrel =  conf_.getParameter<edm::ParameterSet>("TH1TTCluster_Barrel");
+  HistoName = "NClusters_IMem_Barrel";
+  Cluster_IMem_Barrel = dqmStore_->book1D(HistoName, HistoName,
+      psTTCluster_Barrel.getParameter<int32_t>("Nbinsx"),
+      psTTCluster_Barrel.getParameter<double>("xmin"),
+      psTTCluster_Barrel.getParameter<double>("xmax"));
+  Cluster_IMem_Barrel->setAxisTitle("Barrel Layer", 1);
+  Cluster_IMem_Barrel->setAxisTitle("# L1 Clusters", 2);
+  
+  HistoName = "NClusters_OMem_Barrel";
+  Cluster_OMem_Barrel = dqmStore_->book1D(HistoName, HistoName,
+      psTTCluster_Barrel.getParameter<int32_t>("Nbinsx"),
+      psTTCluster_Barrel.getParameter<double>("xmin"),
+      psTTCluster_Barrel.getParameter<double>("xmax"));
+  Cluster_OMem_Barrel->setAxisTitle("Barrel Layer", 1);
+  Cluster_OMem_Barrel->setAxisTitle("# L1 Clusters", 2);
+  
+  edm::ParameterSet psTTCluster_ECDisc =  conf_.getParameter<edm::ParameterSet>("TH1TTCluster_ECDiscs");
+  HistoName = "NClusters_IMem_Endcap_Disc";
+  Cluster_IMem_Endcap_Disc = dqmStore_->book1D(HistoName, HistoName,
+      psTTCluster_ECDisc.getParameter<int32_t>("Nbinsx"),
+      psTTCluster_ECDisc.getParameter<double>("xmin"),
+      psTTCluster_ECDisc.getParameter<double>("xmax"));
+  Cluster_IMem_Endcap_Disc->setAxisTitle("Endcap Disc", 1);
+  Cluster_IMem_Endcap_Disc->setAxisTitle("# L1 Clusters", 2);
+  
+  HistoName = "NClusters_OMem_Endcap_Disc";
+  Cluster_OMem_Endcap_Disc = dqmStore_->book1D(HistoName, HistoName,
+      psTTCluster_ECDisc.getParameter<int32_t>("Nbinsx"),
+      psTTCluster_ECDisc.getParameter<double>("xmin"),
+      psTTCluster_ECDisc.getParameter<double>("xmax"));
+  Cluster_OMem_Endcap_Disc->setAxisTitle("Endcap Disc", 1);
+  Cluster_OMem_Endcap_Disc->setAxisTitle("# L1 Clusters", 2);
+  
+  edm::ParameterSet psTTCluster_ECRing =  conf_.getParameter<edm::ParameterSet>("TH1TTCluster_ECRings");  
+  HistoName = "NClusters_IMem_Endcap_Ring";
+  Cluster_IMem_Endcap_Ring = dqmStore_->book1D(HistoName, HistoName,
+      psTTCluster_ECRing.getParameter<int32_t>("Nbinsx"),
+      psTTCluster_ECRing.getParameter<double>("xmin"),
+      psTTCluster_ECRing.getParameter<double>("xmax"));
+  Cluster_IMem_Endcap_Ring->setAxisTitle("Endcap Ring", 1);
+  Cluster_IMem_Endcap_Ring->setAxisTitle("# L1 Clusters", 2);
+  
+  HistoName = "NClusters_OMem_Endcap_Ring";
+  Cluster_OMem_Endcap_Ring = dqmStore_->book1D(HistoName, HistoName,
+      psTTCluster_ECRing.getParameter<int32_t>("Nbinsx"),
+      psTTCluster_ECRing.getParameter<double>("xmin"),
+      psTTCluster_ECRing.getParameter<double>("xmax"));
+  Cluster_OMem_Endcap_Ring->setAxisTitle("Endcap Ring", 1);
+  Cluster_OMem_Endcap_Ring->setAxisTitle("# L1 Clusters", 2);
+  
+  for(int i=0;i<5;i++){
+    Char_t histo[200];
+    sprintf(histo, "NClusters_IMem_Disc+%d", i+1);  
```

Comment:
instead of Char_t[200] please use std::string here and below. std::to_string will convert the integer.